File: Week 1 - C/2-mario-less.c
#include <stdio.h>

int main(void)
{
    int rows, i, j, spaces;

    while (1)
    {
        printf("Enter number of rows (greater than 0): ");
        scanf("%d", &rows);

        if (rows > 0)
        {
            break;
        }
        else
        {
            printf("Invalid input. Please enter a number greater than 0.\n");
        }
    }

    for (i = 1; i <= rows; i++)
    {
        // Print spaces
        for (spaces = 1; spaces <= rows - i; spaces++)
        {
            printf(" ");
        }

        // Print '#' symbols
        for (j = 1; j <= i; j++)
        {
            printf("#");
        }

        printf("\n");
    }

    return 0;
}

File: Week 6 - Python/2-mario-less.py
def get_height():
    while True:
        try:
            height = int(input("Enter the height (1-8): "))
            if 1 <= height <= 8:
                return height
            else:
                print("Please enter a number between 1 and 8.")
        except ValueError:
            print("Invalid input. Please enter a numeric value.")


def print_triangle(height):
    for i in range(1, height + 1):
        print(" " * (height - i) + "#" * i)


if __name__ == "__main__":
    height = get_height()
    print_triangle(height)